### DISCUSSION TITLE
Application Logs not visible after 3b796163591a300a433e25384f2574419e51da1a

### DISCUSSION THREAD

**Discussion by sanchay9** [mfussenegger/nvim-dap#1407] [General]
Breakpoints and debugging works, but logs are not visible in dap-repl buffer.

<img width="2057" alt="image" src="https://github.com/user-attachments/assets/d785c48f-706c-4a0c-8ae4-1a1ab0c08fbc" />

**Comment by sanchay9** [comment]
upd: I can see the logs divided into stderr and stdout log files in .cache/nvim on :DapShowLog

how can I have all these in dap-repl buffer just like before.

**Reply by mfussenegger** [reply]
What's the use-case for seeing the stderr/stdout of the debug adapter in the REPL?

Any output from debug adapters that's supposed to be visible to the end-user should go through output events.

**Reply by zivkovicn** [reply]
![Screenshot 2024-12-30 at 10 49 17 PM](https://github.com/user-attachments/assets/ecd97774-9328-4867-99c5-37b4e7ba0b35)
@mfussenegger I think that @sanchay9 mean to say, he was expecting to see the logs in the console pane (right of the REPL pane) in dap ui

**Reply by mfussenegger** [reply]
Showing an empty window isn't telling me much but I think I can put the puzzle pieces together now.

It looks like `delve` by default doesn't use output events, so you were relying on stdout/stderr being sent to the REPL to see the stdout of your application. nvim-dap so far only did that as a workaround.

But there's a `outputMode = "remote"` option for delve to get it to use output events.

E.g.:

```lua
dap.configurations.go = {
  {
    type = "delve",
    name = "file",
    request = "launch",
    program = "${file}",
    outputMode = "remote",
  }
}
```

See https://github.com/go-delve/delve/pull/3253

**Reply by sanchay9** [reply]
yes, that was the case, after adding remote option, I can see the logs again. Appreciate the detailed explanation, thanks!
